src/ipmanager/api/views.py

```diff
@@ -43,4 +44,55 @@ def get(self, request, group_key):
                 "status": 404
             }
 
-            return JsonResponse(body, status=404)
\ No newline at end of file
+            return JsonResponse(body, status=404)
+        
+
+class APICheckView(View):
+
+  def get(self, request):
+
+    # Made-up data
+    group_keys = ["A", "B", "C", "D"]
+
+    # Using query_string (unparsed) and to create Query Dict object to extrack parameters
+    dict = QueryDict(query_string = request.META['QUERY_STRING'], mutable=False, encoding=None)
+    
+    # Returning 400 error if wrong parameters or ip not present
+    if 'ip' not in dict:
+      return JsonResponse({"title": "Unparseable IP address", 
+                           "detail": "The value 'invalid_ip' you provided in the 'ip' query parameter cannot be" + 
+                                     "parsed as a valid IP address.", 
+                           "status": 400}, status=400)
+    
+    # Can extract ip from dict now
+    ip = dict['ip']
+
+    if 'group' not in dict:
+      return JsonResponse({"@id":request.build_absolute_uri(), 
+                           "ip":ip,
+                           # Demo-only data, with actual data iterate through all group_keys and make url and parse response accordingly
+                           "checks":[{"@id":"https://ipmanager-test.lib.umd.edu/check?group=Henson00001&ip=127.0.0.1",
+                                      "group":{"@id":"ttps://ipmanager-test.lib.umd.edu/groups/Henson00001",
```

Comment:
I know this is just placeholder data, but you are missing an "h" at the beginning of the group `@id`.